### PR TITLE
tests/xtimer: add small values test for xtimer_usleep

### PR DESCRIPTION
**Pull request by MichelRottleuthner**
On some boards xtimer_usleep never returns when called with rather small values. 
This adds a small test that shows if the problem exists on the used platform. 
On boards where this test fails (e.g `nucleo-l1`) the xtimer config ([`XTIMER_BACKOFF `](https://github.com/RIOT-OS/RIOT/blob/master/boards/nucleo-l1/include/board.h#L37)?)  probably needs to be adjusted. See #7347 for more details.

### CODE REVIEW

File: tests/xtimer_usleep_short/main.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    xtimer_sleep(3);
+    printf("This test will call xtimer_usleep for values from %d down to %d\n",
+            TEST_USLEEP_MAX, TEST_USLEEP_MIN);
+
+    for (int i = TEST_USLEEP_MAX; i >= TEST_USLEEP_MIN; i--) {
+        printf("going to sleep %d usecs...\n", i);
+        xtimer_usleep(i);
+    }
+
+    puts("Test end.");
```
Comment (smlng) [inline tests/xtimer_usleep_short/main.c]: we mostly use something like `[SUCCESS]`, which I find more fitting and explicit.

File: tests/xtimer_usleep_short/main.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#define TEST_USLEEP_MIN (0)
+#define TEST_USLEEP_MAX (500)
+
+int main(void)
+{
+    xtimer_sleep(3);
+    printf("This test will call xtimer_usleep for values from %d down to %d\n",
+            TEST_USLEEP_MAX, TEST_USLEEP_MIN);
+
+    for (int i = TEST_USLEEP_MAX; i >= TEST_USLEEP_MIN; i--) {
```
Comment (smlng) [inline tests/xtimer_usleep_short/main.c]: I think its better (and fine) to check `i > TEST_USLEEP_MIN`.

File: tests/xtimer_usleep_short/tests/01-run.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    i = 500
+
+    while (i >= 0):
+        try:
+            child.expect(u"going to sleep \\d+ usecs...\r\n", timeout=3)
+        except pexpect.TIMEOUT:
+            print("xtimer stuck when trying to sleep %d usecs" % (i+1));
+            print("[FAILED]")
+            break;
+        i = i -1
```
Comment (smlng) [inline tests/xtimer_usleep_short/tests/01-run.py]: spacing 
```
i = i - 1
       ^
```

File: tests/xtimer_usleep_short/tests/01-run.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+sys.path.append(os.path.join(os.environ['RIOTBASE'], 'dist/tools/testrunner'))
+import testrunner
+
+def testfunc(child):
+
+    child.expect(u"This test will call xtimer_usleep for values from \\d+ down to \\d+\r\n")
+
+    i = 500
+
+    while (i >= 0):
```
Comment (smlng) [inline tests/xtimer_usleep_short/tests/01-run.py]: if you change the loop in `main.c` this must be `i > 0`, too

File: tests/xtimer_usleep_short/main.c (hunk 2 above)
Comment (MichelRottleuthner) [inline tests/xtimer_usleep_short/main.c]: what about xtimer_usleep(0)? Yes, in theory that wouldn't make sense, but nonetheless in some applications the value handed to xtimer_usleep may be calculated and possibly is zero. I think the timer to sleep for zero usecs should't lead to a lockup.

Comment (smlng) [inline tests/xtimer_usleep_short/main.c]: ok, fine by me - leave as is

**Comment by smlng** [comment]
tested on `pba-d-01-kw2x` and native on macOS, works - so ACK and GO!

**Pull request merged**